src/test/clojure/clara/test_rules.clj

```diff
@@ -907,6 +907,41 @@
            (set batch-inserts)
            (set single-inserts)))))
 
+(def test-accum-needing-token-partitions-correctly-on-fact-binding
+  (let [q (dsl/parse-query []
+                           [[WindSpeed]
+                            [?ts <- (acc/all) :from [Temperature (= ?loc location)]]])
```

Comment:
I've updated this test so that the AccumulateWithJoinFilterNode is also tested for this scenario.